scripts/globals/abyssea.lua

```diff
@@ -399,3 +415,170 @@ function abysseaOnEventFinish(player,csid,option)
         return true
     end
 end
+
+local abyssea_zones =
+{
+    15, -- ABYSSEA_KONSCHTAT
+    45, -- ABYSSEA_TAHRONGI
+    132,-- ABYSSEA_LA_THEINE
+    215,-- ABYSSEA_ATTOHWA
+    216,-- ABYSSEA_MISAREAUX
+    217,-- ABYSSEA_VUNKERL
+    218,-- ABYSSEA_ALTEPA
+    253,-- ABYSSEA_ULEGUERAND
+    254 -- ABYSSEA_GRAUBERG
+    --255 -- ABYSSEA_EMPYREAL_PARADOX
+}
+
+function isInAbysseaZone(player)
+    for index, zone in ipairs(abyssea_zones) do
+        if zone == player:getZoneID() then
+            return true
+        end
+    end
+    return false
+end
+
+function ResetPlayerLights(player)
+    player:setCharVar("pearlLight", 0 + ABYSSEA_BONUSLIGHT_AMOUNT)
+    player:setCharVar("azureLight", 0 + ABYSSEA_BONUSLIGHT_AMOUNT)
+    player:setCharVar("rubyLight",  0 + ABYSSEA_BONUSLIGHT_AMOUNT)
+    player:setCharVar("amberLight", 0 + ABYSSEA_BONUSLIGHT_AMOUNT)
+    player:setCharVar("goldLight",  0 + ABYSSEA_BONUSLIGHT_AMOUNT)
+    player:setCharVar("silverLight",0 + ABYSSEA_BONUSLIGHT_AMOUNT)
+    player:setCharVar("ebonLight",  0 + ABYSSEA_BONUSLIGHT_AMOUNT)
+end
+
+function AddPlayerLights(player, light, amount)
+    local zoneId = player:getZoneID()
+    local ID = zones[zoneId]
+
+    local tiermsg     = 0
+    local lightamount = amount or 0
+    local pearl       = player:getCharVar("pearlLight")
+    local azure       = player:getCharVar("azureLight")
+    local ruby        = player:getCharVar("rubyLight")
+    local amber       = player:getCharVar("amberLight")
+    local gold        = player:getCharVar("goldLight")
+    local silver      = player:getCharVar("silverLight")
+    local ebon        = player:getCharVar("ebonLight")
```

Comment:
It looks like these are only being used in their respective blocks below, can we move these definitions down there so we only load the ones we need?

Reply:
Why chavars when any zoning loses it on retail? setLocalVar() would fit with retail behavior.

Reply:
Retail behavior saves lights on logout.

Reply:
..has it always been that way? Old man brain _swears_ I always lost them if I dc’d

Reply:
Not sure, wikia has this line from 2011,
> Built-up lights are only lost when a character leaves Abyssea. Changing to a different enemy target or logging out/disconnecting does not remove lights, nor do they decay over time.

Reply:
Sounds like we could potentially store them in a hidden status effect and then flag the effect to be silently lost on zonage - like the Prowess buffs.

Speaking of effects, past me likely did the cruor buffs incorrectly even though they work. >.>;;  (used dummy IDs)

Reply:
hmm, i may have to look into what Teo suggested and add them to visitant status effect maybe?

Reply:
oh, thinking about it, i may have to store this in a sql, ugh.

Reply:
Status effects and char vars are both stored in the db, char vars read the db every time you use them, so if you keep using them then just make sure you're doing it efficiently as you can.

Reply:
You would likely need a different status effect for each light type. You could theoretically fit 3 light types into 1 char var, if you wanna go into that.

Reply:
i'll think about it tommorow, after i get some sleep, cheers

Reply:
even if you used every parameter addStatusEffect has I don't think you could cram it all into visitant. so 7 effects or 7 charVars I guess, that you then have to remember to kill anytime player leaves zone in anyway, which makes me think effect route is less effort.

Reply:
after disscussing this with others, it seems, the charVar method will be best suited for this task, so this should not be an issue now that the lights should be removed like retail, they stay on death, logout, onlt reset on zone or timeout